devicetypes/smartthings/fidure-thermostat.src/fidure-thermostat.groovy

```diff
@@ -805,32 +807,31 @@ def lock()
   //log.debug "current lock is: ${val}"
 
   if (val == "00")
-      val = getLockMap().find { it.value == (settings.lock_level ?: "Full") }?.key
+      val = getLockMap().find { it.value == (settings.lock_level ?: "Unlocked") }?.key
   else
       val = "00"
 
- "st rattr 0x${device.deviceNetworkId} 1 0x204 0x01"
-
+  ["st wattr 0x${device.deviceNetworkId} 1 0x204 1 0x30 {${val}}",
```

Comment:
use the zigbee abstraction commands to send raw commands. http://docs.smartthings.com/en/latest/ref-docs/zigbee-ref.html#zigbee-ref